Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -1,37 +1,284 @@
+                        catch (FormatException)
+                        {
+                            Console.WriteLine($"{allAmounts[j + 1]} is not a  valid number");
+                            break;
+                        }
+                    }
+                    catch (FormatException)
+                    {
+                        Console.WriteLine($"{allAmounts[j]} is not a valid number");
+                        break;
+                    }
+                }
+            }
+
+            if (PersonWithHighestLoss.ToString().Contains(','))
+            {
+               ReplaceCommaWithAnd(PersonWithHighestLoss);
+            }
+
+
+
+            return $"{PersonWithHighestLoss} lost the most money. -¤{HighestLossEver}.";
         }
 
         /// <summary>
         /// Return name and current money of the richest person.
         /// </summary>
         public static string FindRichestPerson(string[] peopleAndBalances)
         {
-            return "";
+            if (IsInValidValidInput(peopleAndBalances))
+            {
+                return InValidOutput;
+            }
+
+            var richestPerson = new StringBuilder();
+            var highestBalance = FindRichest(peopleAndBalances, richestPerson);
+
+            if (richestPerson.ToString().Contains(", "))
+            {
+                ReplaceCommaWithAnd(richestPerson); 
+                return $"{richestPerson.ToString()} are the richest people. ¤{highestBalance}.";
+            }
+            else
+            {
+                return $"{richestPerson.ToString()} is the richest person. ¤{highestBalance}.";
+            }
+        }
+
+        private static decimal FindRichest(string[] peopleAndBalances, StringBuilder richestPerson)
+        {
+            var highestBalance = decimal.MinValue;
+            for (int i = 0; i <= peopleAndBalances.Length - 1; i++)
+            {
+                var currentPersonData = peopleAndBalances[i].Split(',');
+                try
+                {
+                    var highestAmountOfPerson = decimal.Parse(currentPersonData[^1]);
+
+                    if (highestAmountOfPerson > highestBalance)
+                    {
+                        highestBalance = highestAmountOfPerson;
+                        richestPerson.Clear(); 
+                        
+                    }
+
+                    if (highestAmountOfPerson >= highestBalance)
+                    {
+                        AddPerson(currentPersonData[0], richestPerson);
+                    }
+                    
+                    
+                }
+                catch (FormatException)
+                {
+                    Console.WriteLine("There was a person with no balance in the file");
+                    break;
+                }
+            }
+
+            return highestBalance;
+        }
+
+        private static StringBuilder AddPerson(string person, StringBuilder sb)
+        {
+            if (sb.Length == 0)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Use the '?:' operator here.](https://app.codacy.com/manual/Almantask/CSharp-From-Zero-To-Hero/pullRequest?prid=4967975)